### ISSUE TITLE
Distribution date banner should re-appear as month nears end

### ISSUE THREAD

**Issue by taoeffect** [okTurtles/group-income#1318]
### Problem

The banner in `GroupDashboard.vue` currently stays hidden if the user dismisses it:

<img width="865" alt="Screen Shot 2022-06-26 at 7 28 59 PM" src="https://user-images.githubusercontent.com/138706/175848936-3962a642-6cb6-42db-ac53-cd589b37c2c4.png">

However, even if their income details are set, it should re-appear 7 days before the end of the current distribution period.

### Solution

Update the logic to make sure it re-appears without breaking the ability to dismiss it once seen (or once income details are added).

**Comment by SebinSong** [comment]
Hi @taoeffect ,

just been looking into some bug tickets for a purpose of becoming more familiar with the code-base, and thought I would need more clarification on this issue.

Apparently, currently the banner only gets displayed if all below three conditions are met.

cond 1] the user hasn't set the income-detail yet.
cond 2] it's within 7 days of income distribution date.
cond 3] the user has dismissed the banner before.


what exactly is the change we want to make to this logic?
pls correctly me if I'm wrong but the issue description sounds like we want to remove the [condition 1] from the logic.

So basically we will display the banner if it nears to the distribution date **"and"** user hasn't already dismissed the banner before, **"regardless of"** whether the income details have been set or not.

Am I correct here? cheers.

**Comment by taoeffect** [comment]
>  pls correctly me if I'm wrong but the issue description sounds like we want to remove the [condition 1] from the logic.

I think [condition 1] may be useful in the sense that if they haven't entered their income details, we definitely want to display it. And if they have updated their income details *within the 7 days of the next distribution date*, then the banner gets dismissed.

> what exactly is the change we want to make to this logic?

If the user has previously dismissed the banner in the previous distribution cycle, but now it is a new one and there are 7 days left until the next distribution date, the banner should re-appear. And it can be dismissed in one of two ways: they update their income details, or they click the X to dismiss the banner.

**Comment by SebinSong** [comment]
Thanks @taoeffect for clarification.

worked on fixing it and just pushed a PR [here](https://github.com/okTurtles/group-income/pull/1322) .

The updated behaviour there is going to be,

if the user has dismissed the banner without updating the income-details, the app won't show the banner until the current distribution cycle ends (existing behaviour)

Then time passes and the user gets into the next distribution cycle, and still the income-details hasn't been updated when it nears the end of the period, then the banner will be shown again. (new behaviour added by my fix)

pls let me know if you spot any issue while reviewing the PR.
